### PR TITLE
Add mercari/grpc-federation plugin v0.9.2

### LINKED ISSUE #991: Plugin request for Buf Schema Registry - grpc-federation
## Mandatory

**Where is the source code for the plugin?**

`protoc-gen-grpc-federation` plugin is found here:

https://github.com/mercari/grpc-federation/tree/main/cmd/protoc-gen-grpc-federation

## Optional

**Does the plugin have a valid semver version?**

We release based on semantic versioning. 

https://github.com/mercari/grpc-federation/releases

What is the latest version, and where did you get this value from?

The latest release is [v0.9.2](https://github.com/mercari/grpc-federation/releases/tag/v0.9.2)

**Does the plugin have a dependency on another plugin?**

gRPC Federation depends on `buf.build/googleapis/googleapis`
https://github.com/mercari/grpc-federation/blob/main/proto/buf.yaml

In addition, `protoc-gen-go` and `protoc-gen-go-grpc` are required to run the code generated by the `protoc-gen-grpc-federation` plugin.

gRPC Federation has made many samples in the repository. Here is the definition of one of them.

- https://github.com/mercari/grpc-federation/tree/main/_examples/01_minimum
- [buf.gen.yaml](https://github.com/mercari/grpc-federation/blob/main/_examples/01_minimum/buf.gen.yaml)

**Does the output of this plugin depend on any other external libraries?**

Here is an example of code generated using `protoc-gen-grpc-federation` .
- https://github.com/mercari/grpc-federation/blob/main/_examples/01_minimum/federation/federation_grpc_federation.pb.go

This code uses the following third party libraries.

- [github.com/google/cel-go/cel](https://pkg.go.dev/github.com/google/cel-go/cel)
- [github.com/google/cel-go/common/types](https://pkg.go.dev/github.com/google/cel-go/common/types)
- [github.com/mercari/grpc-federation/grpc/federation](https://pkg.go.dev/github.com/mercari/grpc-federation/grpc/federation)
- [github.com/mercari/grpc-federation/grpc/federation/cel](https://pkg.go.dev/github.com/mercari/grpc-federation/grpc/federation/cel)
- [go.opentelemetry.io/otel](https://pkg.go.dev/go.opentelemetry.io/otel)
- [go.opentelemetry.io/otel/trace](https://pkg.go.dev/go.opentelemetry.io/otel/trace)
- [google.golang.org/grpc/codes](https://pkg.go.dev/google.golang.org/grpc/codes)
- [google.golang.org/grpc/status](https://pkg.go.dev/google.golang.org/grpc/status)

### PR DESCRIPTION
**Pull request by mfridman**
Fix #991

### CODE REVIEW

File: plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: v1
+name: buf.build/community/mercari-grpc-federation
```
Comment (mfridman) [inline plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml:2]: cc @goccy

This plugin will be available at buf.build/community/mercari-grpc-federation and can be invoked in any buf.gen.yaml file using `buf generate`.

File: plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+plugin_version: v0.9.2
+source_url: https://github.com/mercari/grpc-federation
+integration_guide_url: https://github.com/mercari/grpc-federation/blob/main/docs/getting_started.md
+description: Generates a gRPC server by writing a custom option in Protocol Buffers
```
Comment (pkwarren) [inline plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml:6]: I know this is from the GitHub description, but is there a better one we could use to better describe what this plugin does?

**Review by pkwarren — APPROVED** [review]

File: plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml (hunk 2 above)
Comment (mfridman) [reply plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml:6]:
```suggestion
description: gRPC Federation automatically generates a BFF (Backend for frontend) server that aggregates and returns the results of gRPC protocol-based microservices.
```

File: plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml (hunk 1 above)
Comment (goccy) [reply plugins/community/mercari-grpc-federation/v0.9.2/buf.plugin.yaml:2]: @mfridman Thank you for your support !! 😄

**Pull request merged**